scripts/checkpatch_inc.sh

```diff
@@ -13,15 +13,25 @@ _CP_EXCL=$(for p in $CHECKPATCH_IGNORE; do echo ":(exclude)$p" ; done)
 
 function _checkpatch() {
 		# Use --typedefsfile if supported by the checkpatch tool
-		typedefs_opt="--typedefsfile typedefs.checkpatch"
-		$CHECKPATCH --help 2>&1 | grep -q -- --typedefsfile || \
-				typedefs_opt="";
+		# AND this file is present
+		typedefs_opt="";
+		if $CHECKPATCH --help 2>&1 | grep -q -- --typedefsfile &&
+		    [ -r typedefs.checkpatch ]; then
+			typedefs_opt="--typedefsfile typedefs.checkpatch"
+		fi
+		# Use --root if supported by the checkpatch tool.
+		rootdir_opt=""
+		if $CHECKPATCH --help 2>&1 | grep -q -- --root; then
+		    local rootdir="${KERNEL_SRC_ROOT:-'.'}"
```

Comment:
Where does this variable come from?

Reply:
You can set this var outside of checkpatch. Generally, you don't need to set up this variable at all, the default value '.' will be used.

Reply:
Wouldn't it be more useful to figure out the kernel source root based on the path to checkpatch.pl instead?

Reply:
There are a lot of situations when it is good to use checkpatch.pl that is not a part of the target kernel.
In my case, I use this script for several different kernel versions, including versions with completely broken checkpatch.pl script. So I have to keep and use one working system-wide copy of the checkpatch.pl.

Reply:
Probably, it is better to use `git rev-parse --show-toplevel` instead "." as default directory.

Reply:
> There are a lot of situations when it is good to use checkpatch.pl that is not a part of the target kernel.

You could just add the `checkpatch.pl` folder to your `$PATH`. I do it all the time.

Reply:
That is exactly what I do :)

Reply:
> > There are a lot of situations when it is good to use checkpatch.pl that is not a part of the target kernel.
> 
> You could just add the `checkpatch.pl` folder to your `$PATH`. I do it all the time.

That is exactly what I do :)

Reply:
So doing a bit of work on `which $CHECKPATCH` should tell where the kernel source tree is located.

Reply:
No. Ok, again.
There's a lot of kernel versions, some of them have broken checkpatch.pl. So I personally keep fresh checkpatch.pl and other related files in my OWN directory (say, ~/bin) and this directory is in $PATH.
So assuming that the kernel source tree depends on checkpatch.pl location can lead to mistakenly use the wrong kernel tree location.

Reply:
To be the clearest: there is NOT the ONLY option to use checkpatch.pl from the current kernel source tree.

Reply:
As you may have noted I'm a bit hesitant to introduce yet another environment variable used to control what make does.

How important is this `--root` switch?
When is it needed (I've never noticed that I've needed it)?
How about setting CHECKPATCH="checkpatch.pl --root ...`?